coordinator/src/orderbook/mod.rs

```diff
@@ -30,120 +41,317 @@ pub mod websocket;
 #[cfg(test)]
 mod tests;
 
-pub struct Orderbook {
+struct Orderbook {
     pool: Pool<ConnectionManager<PgConnection>>,
-    shorts: Shorts,
-    longs: Longs,
+    shorts: OrderbookSide,
+    longs: OrderbookSide,
+    // TODO(holzeis): Split up order matching fee rate into taker and maker fees.
+    order_matching_fee_rate: Decimal,
 }
 
-pub struct Shorts(Vec<Order>);
+struct OrderbookSide {
+    orders: Vec<Order>,
+}
 
-impl Shorts {
-    pub fn add(&mut self, order: Order) {
-        self.0.push(order);
+impl OrderbookSide {
+    pub fn new() -> Self {
+        Self { orders: vec![] }
     }
-    pub fn remove(&mut self, order_id: Uuid) {
-        self.0.retain(|o| o.id != order_id)
+    /// adds the given order to the orderbook.
+    pub fn add_order(&mut self, order: Order) {
+        self.orders.push(order);
+        self.sort(order.direction);
     }
-    pub fn sort(&mut self) {
-        self.0.sort_by(|a, b| {
-            if a.price.cmp(&b.price) == Ordering::Equal {
-                return a.timestamp.cmp(&b.timestamp);
-            }
-            // Descending order.
-            b.price.cmp(&a.price)
-        })
+    /// updates an order in the orderbook. Ignores the update if order does not exist.
+    pub fn update_order(&mut self, updated_order: Order) {
+        if let Some(order) = self.orders.iter_mut().find(|o| o.id == updated_order.id) {
+            *order = updated_order;
+        }
+        self.sort(updated_order.direction);
```

Comment:
Imho we can assume that the rate never changes and hence we don't need to resort the list after an update.

Reply:
can you elaborate on that? Why shouldn't I update the price of a limit order?

Reply:
I would not allow this and only allow for changing the quantity. Updating the price makes things more complicated. E.g. on BitMEX this is also not possible.

Reply:
Happy to change that, but I don't see the complexity. Can you explain?

Reply:
it's a simplification: 
- no need to resort lists on the server side nor on the client side if the order was updated
- what do you do if the order was already partially matched?

Reply:
> - no need to resort lists on the server side nor on the client side if the order was updated

I switched the implementation to a BTreeMap so the sorting is inherent to the data structure.

> - what do you do if the order was already partially matched?

Hmm.. that would be indeed a bit more complex. 

We would need to add a check that only as much quantity can be reduced as still open.

Other than that I guess I would simply update the order. But maybe we don't do updates at all and simply delete and insert, like @luckysori suggested?

Reply:
> Other than that I guess I would simply update the order. But maybe we don't do updates at all and simply delete and insert, like @luckysori suggested?

yes, I would say so.